app/mailers/intervention_mailer.rb

```diff
@@ -19,4 +19,24 @@ def invite_to_intervention_and_registration(intervention, email, health_clinic =
 
     mail(to: @email, subject: I18n.t('intervention_mailer.invite_to_intervention_and_registration.subject'))
   end
+
+  def share_externally_and_registration(intervention, email)
+    @intervention = intervention
+    @email = email
+    @user = User.find_by(email: email)
+    @user.send(:generate_invitation_token!) # same as in session mailer
+    @invitation_token = @user.raw_invitation_token
+
+    mail(to: @email, subject: I18n.t('intervention_mailer.share_externally_and_registration.subject'))
+  end
+
+  def share_question_group_and_registration(intervention, email)
+    @intervention = intervention
+    @email = email
+    @user = User.find_by(email: email)
+    @user.send(:generate_invitation_token!) # same as in session mailer
+    @invitation_token = @user.raw_invitation_token
+
+    mail(to: @email, subject: I18n.t('intervention_mailer.share_question_group_and_registration.subject'))
+  end
```

Comment:
To be honest I wonder if we really need these two separate methods if the email literally looks the same 🤔
Additionally,  when you trigger `share_question_group_and_registration` method, in the email you see the name of the original intervention instead of copies. It may be a little confusing when the new researcher login and see the intervention with a different name compering to the email

Reply:
It was mostly a choice, and it could be confusing both ways, as the intervention the invited researcher gets has a different name than the one shared by the inviting researcher. But I can agree that showing the exact filename for the invited researcher could avoid confusion